feat: enhance UniversalPresentationVerifier to allow custom credential verifiers

mohamed-affinidi opened the pull request:
Use `UniversalVerifier` to verify VP credentials, as it is hard to apply custom verifiers to credentials in VP

ex: use `RevocationVerifier` to `vp.verifiableCredential`

mohamed-abdelsamei commented on lib/src/credentials/presentations/verification/vp_integrity_verifier.dart:
```diff
@@ -38,17 +37,6 @@ class VpIntegrityVerifier implements VpVerifier {
       );
     }
-    // Create instance of [VcIntegrityVerifier] for credential-level integrity checks.
-    final vcIntegrityVerifier =
```
Thanks @cosminm-affnd, Agree, it makes more sense for VpIntegrityVerifier to perform VcIntegrity check still
Please check the updated version
- Keep VC integrity check inside VP integrity check
- Add option for user to provide `customCredentialVerifiers` to be applied for each VC

mohamed-affinidi replied:
Thanks @cosminm-affnd, Agree, it makes more sense for VpIntegrityVerifier to perform VcIntegrity check still
Please check the updated version

- Keep the VC integrity check inside the VP integrity check
- Add option for user to provide `customCredentialVerifiers` to be applied for each VC

vinuth-m commented on lib/src/credentials/presentations/suites/universal_presentation_verifier.dart:
```diff
@@ -40,28 +46,52 @@ final class UniversalPresentationVerifier {
     }
 
+    // Verify each credential using the default credential verifiers
+    for (final vcVerifier in UniversalVerifier().defaultVerifiers) {
```
We dont need to iterate through the verifiers. We could do something like:

```
final vcVerifier = UniversalVerifier(customVCVerifiers);
for (final credential in vp.verifiableCredential) {
        final vcVerificationResult = await vcVerifier.verify(credential);
        errors.addAll(vcVerificationResult.errors);
        warnings.addAll(vcVerificationResult.warnings);
}
```

When we do this, we no longer need to have vcIntegrityVerifier within VPIntegrityVerifier. And all default VC verifiers along with any customVCVerifiers given will be automatically run by the UniversalVerifier.

vinuth-m replied:
We wont need `VcIntegrityVerifier` within `VpIntegrityVerifier` when we are running all the default verifiers.

mohamed-affinidi commented:
**UniversalPresentationVerifier**
- Updated UniversalPresentationVerifier to use UniversalVerifier for individual credential verification within presentations.
- Added support for custom VC verifiers (customVclVerifiers) to allow more flexible credential-level verification.

**VpIntegrityVerifier** 
- Updated to handle VP-level integrity checks only. Individual credentials within the VP should be verified using VcIntegrityVerifier.
